src/PooledArrays.jl

```diff
@@ -121,6 +135,13 @@ If `array` is not a `PooledArray` then the order of elements in `refpool` in the
 
 Note that if you hold mutable objects in `PooledArray` it is not allowed to modify them
 after they are stored in it.
+
+In order to improve performance of `getindex` and `copyto!` operations `PooledArray`s
+may share `pool` and `invpool` fields. This sharing is automatically handled
+and is removed for any array sharing common pool if new levels are added to it.
```

Comment:
I wonder whether it's a good idea to mention internal fields in the docstring. Also this is just an optimization so maybe it should only be mentioned in docs about implementation details?

Reply:
I think we should as legacy codes, could have used `pool` and `invpool` fields + we mention sharing below. I have rewritten it to make it more less dependent on implementation details.